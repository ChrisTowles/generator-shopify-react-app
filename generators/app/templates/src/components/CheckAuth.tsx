import * as React from "react";
import { Redirect } from "react-router-dom";
import { SHOP_KEY, TOKEN_KEY } from "../constants";

// Check if the user is logged in (session storage has a token and shop). If they don't then redirect to the login
// page, otherwise display the children
export function CheckAuth(props: { children?: any }): JSX.Element {
    const shop = localStorage.getItem(SHOP_KEY);
    const token = localStorage.getItem(TOKEN_KEY);
    if (!token || !shop) {
        return <Redirect push={true} to="/login" />;
    }
    return props.children;
}
