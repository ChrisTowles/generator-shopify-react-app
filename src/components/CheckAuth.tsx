import * as React from "react";
import { Redirect } from "react-router-dom";
import { config } from "../config";

// Check if the user is logged in (session storage has a token and shop). If they don't then redirect to the login
// page, otherwise display the children
export function CheckAuth(props: { children?: any }): JSX.Element {
    const token = localStorage.getItem(config.tokenKey);
    const shop = localStorage.getItem(config.shopKey);
    if (!token || !shop) {
        return <Redirect push={true} to="/login" />;
    }
    return props.children;
}