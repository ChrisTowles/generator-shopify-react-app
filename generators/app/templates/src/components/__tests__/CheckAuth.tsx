import { shallow } from "enzyme";
import * as React from "react";
import * as renderer from "react-test-renderer";

import { CheckAuth } from "../CheckAuth";

test("Renders correctly with no shop", () => {
    const login = shallow(<CheckAuth shop={null} token={"authToken"}><h1>It's Ok</h1></CheckAuth>);

    expect(login).toMatchSnapshot();
});

test("Renders correctly with no token", () => {
    const login = shallow(<CheckAuth shop={"example.myshopify.com"} token={null}><h1>It's Ok</h1></CheckAuth>);

    expect(login).toMatchSnapshot();
});

test("Renders correctly with shop and token", () => {
    const login = <CheckAuth shop={"example.myshopify.com"} token={"authToken"}><h1>It's Ok</h1></CheckAuth>;

    const component = renderer.create(login);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
