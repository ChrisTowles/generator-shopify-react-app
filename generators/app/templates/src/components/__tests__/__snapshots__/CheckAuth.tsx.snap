// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with no shop 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Redirect
    push={true}
    to="/login"
  />,
  "nodes": Array [
    <Redirect
      push={true}
      to="/login"
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CheckAuth
        shop={null}
        token="authToken"
      >
        <h1>
          It's Ok
        </h1>
      </CheckAuth>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <h1>
            It's Ok
          </h1>,
          "shop": null,
          "token": "authToken",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Redirect
          push={true}
          to="/login"
        />,
        "_debugID": 2,
        "_renderedOutput": <Redirect
          push={true}
          to="/login"
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CheckAuth
    shop={null}
    token="authToken"
  >
    <h1>
      It's Ok
    </h1>
  </CheckAuth>,
}
`;

exports[`Renders correctly with no token 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Redirect
    push={true}
    to="/login"
  />,
  "nodes": Array [
    <Redirect
      push={true}
      to="/login"
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CheckAuth
        shop="example.myshopify.com"
        token={null}
      >
        <h1>
          It's Ok
        </h1>
      </CheckAuth>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <h1>
            It's Ok
          </h1>,
          "shop": "example.myshopify.com",
          "token": null,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Redirect
          push={true}
          to="/login"
        />,
        "_debugID": 4,
        "_renderedOutput": <Redirect
          push={true}
          to="/login"
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CheckAuth
    shop="example.myshopify.com"
    token={null}
  >
    <h1>
      It's Ok
    </h1>
  </CheckAuth>,
}
`;

exports[`Renders correctly with shop and token 1`] = `
<h1>
  It's Ok
</h1>
`;
